usr/src/cmd/dd/dd.c

```diff
@@ -1035,6 +1066,12 @@ main(int argc, char **argv)
 
 			ibc = read(ibf, (char *)ibuf, ibs);
 
+			if (istriden > 0 && lseek(ibf, istriden * ((off_t)ibs),
```

Comment:
Nothing; nowhere else in this file do we have significant safeguards against overflow either (see lines 1011, 1019, 899). It would probably be worth implementing such safeguards, though I'm not sure what the correct behavior would be, though I don't know if this bug is the right place to go fix them all. And, in practice, we're much less likely to overflow when multiplying the stride by a blocksize than we are when multiplying seek by blocksize (seek, in my experience, often being much larger).